tests/unit/test_messages_helpers.py

```diff
@@ -127,3 +129,46 @@ def test_getlogpath_ignore_other_files(test_log_dir, monkeypatch):
     new_fpath.touch()
     present_logs = sorted((test_log_dir / "testapp").iterdir())
     assert present_logs == [f_aaa] + previous_fpaths[1:] + [new_fpath, f_zzz]
+
+
+# -- tests for the _Progresser class
+
+
+def test_progresser_absolute_mode():
+    """Just use _Progresser as a context manager in absolute mode."""
+    stream = sys.stdout
+    text = "test text"
+    total = 123
+    fake_printer = MagicMock()
+    with _Progresser(fake_printer, total, text, stream, delta=False) as progresser:
+        progresser.advance(20)
+        progresser.advance(30)
+
+    assert fake_printer.mock_calls == [
+        call.progress_bar(stream, text, 20, total),
+        call.progress_bar(stream, text, 30, total),
+    ]
+
+
+def test_progresser_delta_mode():
+    """Just use _Progresser as a context manager in delta mode."""
+    stream = sys.stdout
+    text = "test text"
+    total = 123
+    fake_printer = MagicMock()
+    with _Progresser(fake_printer, total, text, stream, delta=True) as progresser:
+        progresser.advance(20)
+        progresser.advance(30)
```

Comment:
We could also test for delta values that bring the accumulated total to be out of range, and also using floating point numbers since we allow floats.

Reply:
Same question above.